server/__tests__/e2e/posts.get.test.ts

```diff
@@ -5,27 +5,62 @@ import {
   getMarketplaceURL,
 } from '../utils';
 import { withRootDb } from '../db/helpers';
-import { commit } from './post.mock';
+import { commit, contents } from './post.mock';
 
 describe('/posts GET endpoint', () => {
   it('returns the post by IRI', async () => {
     const { authHeaders } = await createNewUserAndLogin();
 
-    // Create project and post administered by this account
-    const { projectId, iri } = await createProjectAndPost({
+    // Create project and posts administered by this account
+    const { projectId, iri: prevIri } = await createProjectAndPost({
       initAuthHeaders: authHeaders,
       noFiles: true,
+      initPrivacy: 'private',
     });
 
-    const resp = await fetch(`${getMarketplaceURL()}/posts/${iri}`, {
+    const currentResp = await fetch(`${getMarketplaceURL()}/posts`, {
+      method: 'POST',
+      headers: authHeaders,
+      body: JSON.stringify({
+        projectId,
+        privacy: 'public',
+        contents: { ...contents, name: 'current', files: [] },
+      }),
+    });
+    const { iri } = await currentResp.json();
+
+    const nextResp = await fetch(`${getMarketplaceURL()}/posts`, {
+      method: 'POST',
+      headers: authHeaders,
+      body: JSON.stringify({
+        projectId,
+        privacy: 'private',
+        contents: { ...contents, name: 'next', files: [] },
+      }),
+    });
+    const { iri: nextIri } = await nextResp.json();
+
+    // Get post with project admin auth headers
+    const authResp = await fetch(`${getMarketplaceURL()}/posts/${iri}`, {
       method: 'GET',
       headers: authHeaders,
     });
 
-    expect(resp.status).toBe(200);
-    const data = await resp.json();
+    expect(authResp.status).toBe(200);
+    const authData = await authResp.json();
     // returned post contents based on privacy settings tested in unit test for getPostData function
-    expect(data.iri).toEqual(iri);
+    expect(authData.iri).toEqual(iri);
+    expect(authData.prevIri).toEqual(prevIri);
+    expect(authData.nextIri).toEqual(nextIri);
+
+    // Get post with any app user
+    const resp = await fetch(`${getMarketplaceURL()}/posts/${iri}`, {
+      method: 'GET',
+    });
+    const data = await resp.json();
```

Comment:
Maybe add a check to make sure this was a successful request?

```suggestion
    expect(resp.status).toBe(200);
    const data = await resp.json();
    expect(data.iri).toEqual(iri);
```